src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/service/UserSyncServiceTest.kt

```diff
@@ -0,0 +1,146 @@
+    assertThat(stats.results["username2"]?.differences).isEqualTo(
+      "not equal: value differences={email=(user2nomis@digital.justice.gov.uk, user2@digital.justice.gov.uk)}"
+    )
+  }
+
+  @Test
+  fun `sync user that is missing from Nomis`() {
+    val usersFromAuth = listOf(
+      AuthUser("username1", "user1auth@digital.justice.gov.uk"),
+      AuthUser("username2", "user2@digital.justice.gov.uk")
+    )
+
+    val usersFromNomis = listOf(
+      NomisUser("username2", "user2@digital.justice.gov.uk")
+    )
+    whenever(authService.getUsers()).thenReturn(usersFromAuth)
+    whenever(nomisService.getAllUsers()).thenReturn(usersFromNomis)
+
+    val stats = userSyncService.sync()
+    verify(authService).getUsers()
+    verify(nomisService).getAllUsers()
+
+    // Nothing for username2 as there are no differences
+    println(stats.results)
+    assertThat(stats.results.size).isEqualTo(1)
+    assertThat(stats.results["username1"]?.updateType).isEqualTo(SyncDifferences.UpdateType.NONE)
+    assertThat(stats.results["username1"]?.differences).isEqualTo(
+      "not equal: only on right={userName=username1, email=user1auth@digital.justice.gov.uk}"
+    )
+  }
+
+  @Test
+  fun `sync user that is missing from Auth`() {
+    val usersFromAuth = listOf(
+      AuthUser("username1", "user1@digital.justice.gov.uk")
+    )
+
+    val usersFromNomis = listOf(
+      NomisUser("username1", "user1@digital.justice.gov.uk"),
+      NomisUser("username2", "user2nomis@digital.justice.gov.uk")
+    )
+    whenever(authService.getUsers()).thenReturn(usersFromAuth)
+    whenever(nomisService.getAllUsers()).thenReturn(usersFromNomis)
+    val stats = userSyncService.sync()
+    verify(authService).getUsers()
+    verify(nomisService).getAllUsers()
+
+    // Nothing for username1 as there are no differences
+    assertThat(stats.results.size).isEqualTo(1)
+    assertThat(stats.results["username2"]?.updateType).isEqualTo(SyncDifferences.UpdateType.NONE)
+    assertThat(stats.results["username2"]?.differences).isEqualTo(
+      "not equal: only on left={userName=username2, email=user2nomis@digital.justice.gov.uk}"
+    )
+  }
+
+  @Test
+  fun `sync users with multiple differences`() {
+    val usersFromAuth = listOf(
+      AuthUser("username1", "user1auth@digital.justice.gov.uk"),
+      AuthUser("username2", "user2@digital.justice.gov.uk"),
+      AuthUser("username3", "user3auth@digital.justice.gov.uk")
+    )
+
+    val usersFromNomis = listOf(
+      NomisUser("username2", "user2@digital.justice.gov.uk"),
+      NomisUser("username3", "user3@digital.justice.gov.uk")
+    )
+    whenever(authService.getUsers()).thenReturn(usersFromAuth)
+    whenever(nomisService.getAllUsers()).thenReturn(usersFromNomis)
+
+    val stats = userSyncService.sync()
+    verify(authService).getUsers()
+    verify(nomisService).getAllUsers()
+
+    // Nothing for username2 as there are no differences
+    println(stats.results)
```

Comment:
Leave this in on purpose?

Reply:
no - that needs to come out- well spotted